airflow/models/baseoperator.py

```diff
@@ -650,6 +651,7 @@ def deps(self) -> Set[BaseTIDep]:
             NotInRetryPeriodDep(),
             PrevDagrunDep(),
             TriggerRuleDep(),
+            BranchDep(),
```

Comment:
```suggestion
            NotPreviouslySkipped(),
```

There are more ways of skipping than just branches.

Reply:
@ashb  what about calling it `PreviouslySkipped` instead of `NotPreviouslySkipped`?

I'm just worried the negation in the naming can be a little confusion.

Reply:
We have `NotInRetryPeriodDep` already , so `NotPreviouslySkippedDep` would follow -- `PreviouslySkippedDep` would _pass_ if it had been skipped, the opposite of what we want.